src/main/java/com/google/sps/ServerMain.java

```diff
@@ -8,13 +8,49 @@
 import org.eclipse.jetty.webapp.Configuration;
 import org.eclipse.jetty.webapp.WebAppContext;
 import org.eclipse.jetty.webapp.WebInfConfiguration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.time.Instant;
+import java.time.Duration;
+import java.util.Date;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.Timestamp;
 
 /**
  * Starts up the server, including a DefaultServlet that handles static files, and any servlet
  * classes annotated with the @WebServlet annotation.
  */
 public class ServerMain {
 
+  private static void deleteOldEvents() {
+    System.out.println("Deleting events that have occurred more than 1 day in the past");
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    Query<Entity> query = Query.newEntityQueryBuilder().setKind("Event").build();
```

Comment:
This seems a bit inefficient. You're essentially querying all of the events in the database and looking through them. I feel a better way may be to construct a query to query specifically for the events that are out of date and then batch delete them. Databases store data in efficient ways so that querying for this is likely to be much more efficient than the O(N) loop that you have below where N is the number of events in the database.

I realize you may have done this because date is currently represented as a string. So perhaps, you can trying storing date as [java.util.Date](https://docs.oracle.com/javase/8/docs/api/java/util/Date.html) to enable you to query for dates more efficiently (if you can filter for dates even if they're represented as a string, don't worry about it, but I suspect that is not the case).

Here are some examples how you can construct [queries](https://cloud.google.com/datastore/docs/concepts/queries#restrictions_on_queries) (you will likely want to filter)

Reply:
Good point, I've changed this! We were already storing the date as "Date and Time" in datastore so this wasn't difficult to do.